pandas/_libs/missing.pyx

```diff
@@ -278,3 +278,126 @@ cdef inline bint is_null_period(v):
     # determine if we have a null for a Period (or integer versions),
     # excluding np.datetime64('nat') and np.timedelta64('nat')
     return checknull_with_nat(v)
+
+
+# -----------------------------------------------------------------------------
+# Implementation of NA singleton
+
+
+def _create_binary_propagating_op(name):
+    import numbers
+
+    def method(self, other):
+        if isinstance(other, numbers.Number) or other is NA or isinstance(other, str):
+            return NA
+
+        return NotImplemented
+
+    method.__name__ = name
+    return method
+
+
+def _create_unary_propagating_op(name):
+    def method(self):
+        return NA
+
+    method.__name__ = name
```

Comment:
can you add a doc-string for these? (auto-construct is ok)

Reply:
This is only for dunder methods, so not a docstring that users typically see? (eg the Series dunder methods also have no docstring)